components/ws-notification/index.js

```diff
@@ -0,0 +1,110 @@
+import { React, Component } from '../imports';
+import './ws-notification.scss';
+
+//const NOTIFICATIONS_SHOW_LIMIT = 3;
+const DEFAULT_NOTIFICATION_LIFETIME = 5000;
+const DEFAULT_NOTIFICATION_TYPE = 'info';
+
+export default class WSNotification extends Component {
+  constructor() {
+    super();
+    this.state = {
+      notifications: [],
+      timeoutId: null
+    };
+    // Bind functions to this scope
+    this.addNotify = this.addNotify.bind(this);
+    this.closeAllEvent = this.closeAllEvent.bind(this);
+  }
+  componentDidMount() {
+    window.addEventListener('ws-notification-open', this.addNotify);
+    window.addEventListener('ws-notification-close-all', this.closeAllEvent);
+  }
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.notifications.length < this.state.notifications.length) {
+      this.animateIn(this.state.notifications[this.state.notifications.length-1], this.state.notifications.length-1);
+    }
+  }
+  componentWillUnmount() {
+    window.removeEventListener('ws-notification-open');
+    window.removeEventListener('ws-notification-close');
+  }
+  addNotify (event) {
+    let {title, description, type, lifetime} = event.detail;
+    if (typeof lifetime === 'undefined') {
+      lifetime = DEFAULT_NOTIFICATION_LIFETIME;
+    } else if (!lifetime) {
+      lifetime = 2147483647;
```

Comment:
I liked the way before more since error notifications don't need any timeout which costs resources.